rdock-utils/rdock_utils/sdrmsd/sdrmsd.py

```diff
@@ -0,0 +1,173 @@
+import functools
+import logging
+import math
+
+from openbabel import pybel
+
+from .superpose3d import MolAlignmentData, Superpose3D
+from .types import AutomorphismRMSD, CoordsArray, PoseMatchData, SDRMSDData
+
+logger = logging.getLogger("SDRMSD")
+
+
+class SDRMSD:
+    def __init__(
+        self,
+        reference_sdf: str,
+        input_sdf: str,
+        fit: bool,
+        threshold: float,
+        output_filename: str,
+    ) -> None:
+        self.reference_sdf = reference_sdf
+        self.input_sdf = input_sdf
+        self.fit = fit
+        self.threshold = threshold
+        self.output_filename = output_filename
+
+    def run(self) -> None:
+        # Find the RMSD between the crystal pose and each docked pose
+        docked_poses = pybel.readfile("sdf", self.input_sdf)
+        self.display_fit_message()
+        data = SDRMSDData()
+
+        # Read crystal pose
+        crystal_pose = self.get_crystal_pose()
+        crystal_atoms = len(crystal_pose.atoms)
+
+        for i, docked_pose in enumerate(docked_poses, start=1):
+            atoms_number = self.process_docked_pose(docked_pose)
+
+            if atoms_number != crystal_atoms:
+                data.skipped.append(i)
+                continue
+
+            rmsd_result = self.calculate_rmsd(crystal_pose, docked_pose)
+            pose_match_data = PoseMatchData(i, docked_pose, data)
+            self.handle_pose_matching(rmsd_result, pose_match_data)
+
+        if self.output_filename:
+            self.process_and_save_selected_molecules(data)
+
+        if data.skipped:
+            logger.warning(f"SKIPPED input molecules due to the number of atom mismatch: {data.skipped}")
+
+    def get_automorphism_rmsd(self, target: pybel.Molecule, molecule: pybel.Molecule) -> AutomorphismRMSD:
+        """
+        Use Automorphism to reorder target coordinates to match reference coordinates atom order
+        for correct RMSD comparison. Only the lowest RMSD will be returned.
+
+        Returns:
+        If fit=False: bestRMSD	(float)					RMSD of the best matching mapping.
+        If fit=True:  (bestRMSD, molecCoordinates)	(float, npy.array)	RMSD of best match and its molecule fitted coordinates.
```

Comment:
this docstring doesn't match the function anymore, i would update it or, more probably, delete it